### PR TITLE
Bring back RPi3 (and 4) support

### PR DESCRIPTION
**Pull request by Herrie82**
Similar to https://github.com/webOS-ports/meta-rpi-luneos/pull/9

Boots to UI, wifi works, BT works, WAM works. Has some performance issues, but good enough state to merge and build upon.

### CODE REVIEW

**Review by shr-project — APPROVED** [review]
Maybe drop commits 3 and 4 now when not using rpi-sdimg?

**Comment by Herrie82** [comment]
> Maybe drop commits 3 and 4 now when not using rpi-sdimg?

Can do that, but if someone wants to use it in the future (for whatever reason), good to have for reference

**Comment by shr-project** [comment]
> Can do that, but if someone wants to use it in the future (for whatever reason), good to have for reference

and does this reference work now? I'll fix the .rootfs in meta-raspberrypi one and then this duplication won't be needed, right?

**Comment by Herrie82** [comment]
> > Can do that, but if someone wants to use it in the future (for whatever reason), good to have for reference
> 
> and does this reference work now? I'll fix the .rootfs in meta-raspberrypi one and then this duplication won't be needed, right?

Yes it does

**Comment by shr-project** [comment]
https://github.com/agherzan/meta-raspberrypi/pull/1289 should work as well without the duplication.

And https://github.com/shr-project/meta-webosose/commit/32dccc2ade1203dfeebcea94abe91a7eabfc4935 refreshes it from meta-raspberrypi (so that the only difference is virtual/kernel:do_webos_deploy_fixup which you're not using anyway.

So I think it should be dropped from meta-rpi-luneos.

File: classes/sdcard_image-rpi.bbclass [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# Imported from meta-raspberrypi
+# kernel dependency changed to do_webos_deploy_fixup task
```
Comment (shr-project) [inline classes/sdcard_image-rpi.bbclass]: This isn't true right? the only difference is:
https://github.com/agherzan/meta-raspberrypi/pull/1289
and do_webos_deploy_fixup task was removed by "FU" commit now squashed to the next commit.

Comment (Herrie82) [reply classes/sdcard_image-rpi.bbclass]: Cleaned that up now, indeed I removed that one to not rely on all the stuff OSE is doing there

**Comment by Herrie82** [comment]
> [agherzan/meta-raspberrypi#1289](https://github.com/agherzan/meta-raspberrypi/pull/1289) should work as well without the duplication.
> 
> And [shr-project/meta-webosose@32dccc2](https://github.com/shr-project/meta-webosose/commit/32dccc2ade1203dfeebcea94abe91a7eabfc4935) refreshes it from meta-raspberrypi (so that the only difference is virtual/kernel:do_webos_deploy_fixup which you're not using anyway.
> 
> So I think it should be dropped from meta-rpi-luneos.

Can drop it once Andrei merges.

[edit]
Andrei isn't the quickest at times... Still have a Kirkstone one pending for weeks already.
[/edit]

**Comment by shr-project** [comment]
> > [agherzan/meta-raspberrypi#1289](https://github.com/agherzan/meta-raspberrypi/pull/1289) should work as well without the duplication.
> > And [shr-project/meta-webosose@32dccc2](https://github.com/shr-project/meta-webosose/commit/32dccc2ade1203dfeebcea94abe91a7eabfc4935) refreshes it from meta-raspberrypi (so that the only difference is virtual/kernel:do_webos_deploy_fixup which you're not using anyway.
> > So I think it should be dropped from meta-rpi-luneos.
> 
> Can drop it once Andrei merges.
> 
> [edit] Andrei isn't the quickest at times... Still have a Kirkstone one pending for weeks already. [/edit]

It's already merged in meta-raspberrypi and included in webos-ports-setup:
https://github.com/webOS-ports/webos-ports-setup/commit/3b124c66120b3899b9c8b6b26e3b1da672496e44

**Comment by Herrie82** [comment]
> > > [agherzan/meta-raspberrypi#1289](https://github.com/agherzan/meta-raspberrypi/pull/1289) should work as well without the duplication.
> > > And [shr-project/meta-webosose@32dccc2](https://github.com/shr-project/meta-webosose/commit/32dccc2ade1203dfeebcea94abe91a7eabfc4935) refreshes it from meta-raspberrypi (so that the only difference is virtual/kernel:do_webos_deploy_fixup which you're not using anyway.
> > > So I think it should be dropped from meta-rpi-luneos.
> > 
> > 
> > Can drop it once Andrei merges.
> > [edit] Andrei isn't the quickest at times... Still have a Kirkstone one pending for weeks already. [/edit]
> 
> It's already merged in meta-raspberrypi and included in webos-ports-setup: [webOS-ports/webos-ports-setup@3b124c6](https://github.com/webOS-ports/webos-ports-setup/commit/3b124c66120b3899b9c8b6b26e3b1da672496e44)

Dropped, happy to squash some commits if needed.

**Comment by shr-project** [comment]
Do we need bigger /boot from webos .wks?
```
-part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
+part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --fixed-size 80M
```

**Comment by Herrie82** [comment]
> Do we need bigger /boot from webos .wks?
> 
> ```
> -part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
> +part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --fixed-size 80M
> ```

Seems boot is using 45MB according df -h from @ericblade, so yes :)

**Comment by shr-project** [comment]
> > Do we need bigger /boot from webos .wks?
> > ```
> > -part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
> > +part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --fixed-size 80M
> > ```
> 
> Seems boot is using 45MB according df -h from @ericblade, so yes :)

ok, looks like the kernel got much bigger again
```
0       rpi-bootfiles-20220830.stamp
2.0K    boot.scr
2.0K    cmdline.txt
4.0K    fixup4cd.dat
4.0K    fixup_cd.dat
4.0K    uEnv.txt
6.0K    fixup4.dat
8.0K    fixup.dat
10K     fixup4db.dat
10K     fixup4x.dat
10K     fixup_db.dat
10K     fixup_x.dat
16K     uboot.env
36K     config.txt
52K     bcm2711-rpi-4-b.dtb
52K     bcm2711-rpi-400.dtb
52K     bcm2711-rpi-cm4.dtb
52K     bootcode.bin
126K    overlays
558K    kernel8.img
786K    start4cd.elf
786K    start_cd.elf
2.2M    start4.elf
2.9M    start.elf
2.9M    start4x.elf
3.6M    start4db.elf
3.6M    start_x.elf
4.6M    start_db.elf
23M     Image
```

Strange that nobody else sees similar issue with the default wks.

**Comment by ericblade** [comment]
558k for a kernel? my kernel is 22mb.  Whole directory is 43mb out of 70.  

```
total 44134
drwxrwxrwx  1 root root      512 Jan  1  1980  .
drwxr-xr-x 10 root root     4096 Feb 22 22:29  ..
-rwxrwxrwx  1 root root    33555 Apr  6  2011  bcm2710-rpi-3-b.dtb
-rwxrwxrwx  1 root root    34182 Apr  6  2011  bcm2710-rpi-3-b-plus.dtb
-rwxrwxrwx  1 root root    31254 Apr  6  2011  bcm2710-rpi-cm3.dtb
-rwxrwxrwx  1 root root    20720 Apr  6  2011  bcm2837-rpi-3-b.dtb
-rwxrwxrwx  1 root root    52476 Apr  6  2011  bootcode.bin
-rwxrwxrwx  1 root root       67 Apr  6  2011  cmdline.txt
-rwxrwxrwx  1 root root     2607 Apr  6  2011  config.txt
-rwxrwxrwx  1 root root     3171 Apr  6  2011  fixup4cd.dat
-rwxrwxrwx  1 root root     5399 Apr  6  2011  fixup4.dat
-rwxrwxrwx  1 root root     8385 Apr  6  2011  fixup4db.dat
-rwxrwxrwx  1 root root     8387 Apr  6  2011  fixup4x.dat
-rwxrwxrwx  1 root root     3171 Apr  6  2011  fixup_cd.dat
-rwxrwxrwx  1 root root     7262 Apr  6  2011  fixup.dat
-rwxrwxrwx  1 root root    10230 Apr  6  2011  fixup_db.dat
-rwxrwxrwx  1 root root    10232 Apr  6  2011  fixup_x.dat
-rwxrwxrwx  1 root root 22804992 Apr  6  2011  kernel8.img
drwxrwxrwx  1 root root      512 Apr  6  2011  overlays
-rwxrwxrwx  1 root root        0 Apr  6  2011  rpi-bootfiles-20230509~buster.stamp
-rwxrwxrwx  1 root root   805724 Apr  6  2011  start4cd.elf
-rwxrwxrwx  1 root root  3747688 Apr  6  2011  start4db.elf
-rwxrwxrwx  1 root root  2251392 Apr  6  2011  start4.elf
-rwxrwxrwx  1 root root  2998888 Apr  6  2011  start4x.elf
-rwxrwxrwx  1 root root   805724 Apr  6  2011  start_cd.elf
-rwxrwxrwx  1 root root  4819624 Apr  6  2011  start_db.elf
-rwxrwxrwx  1 root root  2975616 Apr  6  2011  start.elf
-rwxrwxrwx  1 root root  3722504 Apr  6  2011  start_x.elf
drwxrwxrwx  1 root root      512 Feb 22 22:22 'System Volume Information'
```

**Comment by shr-project** [comment]
https://github.com/agherzan/meta-raspberrypi/pull/1290 was merged, you can drop LGEs WKS file.

**Comment by Herrie82** [comment]
> [agherzan/meta-raspberrypi#1290](https://github.com/agherzan/meta-raspberrypi/pull/1290) was merged, you can drop LGEs WKS file.

Sorry need more coffee... wanted to PR this, but merged instead https://github.com/webOS-ports/meta-rpi-luneos/commit/a1738d8001cd1a711c86d3f270161255bdb21f55

**Pull request merged**